src/services/StorageService.ts

```
@@ -1,47 +1,30 @@
-      }
-    } catch (error: any) {
-      // Only throw if it's not a permission issue (for tests)
-      if (error.code !== 'EACCES' && error.code !== 'EPERM') {
-        console.error("Failed to create data directory:", error);
-        throw new Error(`Cannot create data directory: ${this.dataDir}`);
-      }
-      console.warn("Directory creation failed, will retry on first operation:", error.message);
-    }
+  private database: ProjectDatabase;
+  private operationLock: boolean = false;
+  private operationQueue: Array<{ resolve: Function, reject: Function, operation: Function }> = [];
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)